hbase_sql/src/main/java/com/splicemachine/derby/impl/kryo/SparkValueRowSerializer.java

```diff
@@ -92,7 +92,7 @@ public void write(Kryo kryo, Output output, T object) {
                         output.writeBoolean(useKryo);
                         ByteArrayOutputStream outputBuffer = new ByteArrayOutputStream();
                         ObjectOutputStream objectOutputStream = new ObjectOutputStream(outputBuffer);
-                        objectOutputStream.writeObject(o);
+                        objectOutputStream.writeUnshared(o);
```

Comment:
I doubt this change is going to make any difference. objectOutputStream is local and becomes collectable once the method returns. There should be no accumulated effect.

Reply:
I thought the issue John was fixing a memory leak issue alleged with ObjectOutputStream. Using ObjectOutputStream.reset() along with writeUnshared()/readUnshared() typically solves the problem.

There are a few related articles on this topics:
https://www.securecoding.cert.org/confluence/display/java/FIO53-J.+Use+the+serialization+methods+writeUnshared%28%29+and+readUnshared%28%29+with+care
http://www.maultech.com/chrislott/blog/20100110_spacetime.html
https://www.reddit.com/r/learnprogramming/comments/66p4be/java_objectinputstream_why_am_i_getting_a_memory/
https://orenkishon.wordpress.com/2014/06/28/java-memory-leak-caused-by-objectinputstream/

Reply:
Those links are OK by themselves but they don't seem to be relevant here as both ObjectOutputStream and UDTInputStream are used for one object only and are gone with all their cached links after serde (no cumulative effect). readUnshared creates more objects (potentially breaking functionality of the deserialized object), thus increasing GC pressure.
I struggle to see why the change could fix any memory leak. If I'm missing something, maybe more details could help here.

Reply:
I'll defer this to John/Yi who investigated the original use case.

Reply:
One thing to try here is to reuse objectOutputStream to avoid the cost of its initialization (quite high in the profile) and reduce the amount of generated garbage. Resetting it after use will be required in that case.